Feature/external dummy herds

andkaha opened the pull request:
Adding dummy herds `GX1` and `MX1` for the two genebanks.  These will hold the animals sold outside of the genebanks (by means of entries added to the `herd_tracking` table).

andkaha committed:
Allow herd names matching [GM]X?[0-9]+

andkaha committed:
Add dummy herds GX1 and MX1

andkaha committed:
Add plan for what the SQL will do

andkaha committed:
Fix SQL syntax error

norling reviewed (changes requested):
If we're going to allow multiple letters in Herd prefixes you need to update the regexps in `frontend/src/user_context.tsx: canEdit()` so that they still trigger. This probably won't be an issue for the dummy herds, but if we allow these herds then the code should reflect that.

andkaha committed:
Make it easier to only load one data set

Leaving out `-g` or `-G` skips loading (and checking) the Gotland data,
while leaving out `-m` skips loading (and checking) the Mellerud data.

This makes it easier to debug the developing of healthchecks (the `data`
table will not be clobbered).

andkaha committed:
Fix comment, add SQL

Rabbits are now added to the dummy herd `GX1` if their "ny G" entry in
the Excel data is "0".

andkaha committed:
Saner names of "dummy herds"

andkaha committed:
Address comment from @norling

Herd names may have an optional `X` character ofter the initial
character and before the number.  This commit modifies a regular
expression to allow for that.

andkaha commented:
> If we're going to allow multiple letters in Herd prefixes you need to update the regexps in `frontend/src/user_context.tsx: canEdit()` so that they still trigger. This probably won't be an issue for the dummy herds, but if we allow these herds then the code should reflect that.

This is addressed in commit  7f164a4.

andkaha commented:
Dummy herds now added, but data for dummy herds is only available for Gotlandskanin as Mellerudskanin does not annotate animals sold to herds outside of the genebank.

jhagberg reviewed (approved):
LGTM good with name "Externa djur"

andkaha committed:
Better date creation

Use `MAKE_DATE()` rather than casting an ad-hoc date string.
